core/spatial_map.cpp

```diff
@@ -2346,28 +2358,27 @@ EidosValue_SP SpatialMap::ExecuteMethod_sampleNearbyPoint(EidosGlobalStringID p_
 						kernel.DrawDisplacement_S1(displaced_point);
 						displaced_point[0] += point_a;
 					}
-					while ((displaced_point[0] < 0.0) || (displaced_point[0] > 1.0));
+					while ((displaced_point[0] < bounds_a0_) || (displaced_point[0] > bounds_a1_));
 				}
 				
-				map_value = ValueAtPoint_S1(displaced_point);
+				// we do most of our work in user-space coordinates; here we scale to [0, 1] for ValueAtPoint
+				double rescaled_point[1];
+				
+				rescaled_point[0] = (displaced_point[0] - bounds_a0_) / (bounds_a1_ - bounds_a0_);
+				map_value = ValueAtPoint_S1(rescaled_point);
 			}
 			while (values_max_ * Eidos_rng_uniform(rng) > map_value);
 			
-			// scale point coordinates back to user space
-			*(result_ptr++) = displaced_point[0] * (bounds_a1_ - bounds_a0_) + bounds_a0_;
+			*(result_ptr++) = displaced_point[0];
 		}
 	}
 	else if (spatiality_ == 2)
 	{
 		// FIXME: TO BE PARALLELIZED
 		for (size_t point_index = 0; point_index < point_count; point_index++)
 		{
-			// scale point coordinates to [0, 1]
 			double point_a = *(point_buf_ptr)++;
```

Comment:
in `sampleImprovedPoint` this was like `double point[X]`, not `point_a`, `point_b`, etcetera. (I don't see anything wrong, just differences in what could be parallel style of code?)

Reply:
Yes.  There was a reason for that, in the old code, but that reason has now evaporated, so I'll shift it to be parallel.